packages/mask/src/components/shared/ApplicationSettingPluginSwitch.tsx

```diff
@@ -63,11 +62,28 @@ const useStyles = makeStyles()((theme) => ({
         color: theme.palette.mode === 'dark' ? theme.palette.text.secondary : theme.palette.text.primary,
     },
 }))
-export function ApplicationSettingPluginSwitch(props: Props) {
+
+interface Props {
+    focusPluginId?: PluginId
+}
+export const ApplicationSettingPluginSwitch = memo(({ focusPluginId }: Props) => {
     const { classes } = useStyles()
     const snsAdaptorPlugins = useActivatedPluginsSNSAdaptor('any')
     const snsAdaptorMinimalPlugins = useActivatedPluginsSNSAdaptor(true)
-    const [open, setOpen] = useState(false)
+    const availablePlugins = useMemo(() => {
+        return snsAdaptorPlugins
+            .flatMap(({ ID, ApplicationEntries: entries }) => (entries ?? []).map((entry) => ({ entry, pluginId: ID })))
+            .filter((x) => x.entry.category === 'dapp')
+            .sort((a, b) => (a.entry.marketListSortingPriority ?? 0) - (b.entry.marketListSortingPriority ?? 0))
+    }, [snsAdaptorPlugins])
+
+    const listRef = useRef<HTMLUListElement>(null)
+    useLayoutEffect(() => {
+        if (!focusPluginId || snsAdaptorPlugins.length === 0 || !listRef.current) return
+        const pluginItem = listRef.current.querySelector(`[data-plugin-id="${focusPluginId}"]`)
```

Comment:
🤔 this is unintended usage... maybe you should provide an extra div to set your own data- property to do this?